engines/myst3/state.cpp

```diff
@@ -531,7 +535,30 @@ bool GameState::load(Common::InSaveFile *saveFile) {
 	return true;
 }
 
-bool GameState::save(Common::OutSaveFile *saveFile) {
+bool GameState::isAutoSaveAllowed() {
+	// Open Autosave.M3S and see if it an autosave
+	// Autosaving will be enabled if Autosave.M3S is an autosave or if Autosave.M3S doesn't exist
+	Common::String fileName = "Autosave.M3S";
+
+	// Check if file exists
+	Common::ScopedPtr<Common::InSaveFile> saveFile(_vm->getSaveFileManager()->openForLoading(fileName));
+	if (!saveFile) { // Autosave.M3S doesn't exist or is corrupt
+		return true;
+	}
+
+	// Get autoSave value from saved file
+	StateData data;
+	Common::Serializer s = Common::Serializer(saveFile.get(), 0);
+	data.syncWithSaveGame(s);
+
+	if (data.autoSave==true) { // Autosave.M3S exists and is an autosave
```

Comment:
Please merge this if statement into a `return <boolean expression>`. Testing `if (something == true)` is considered poor style.

Reply:
Is the change to be:
if (data.autoSave==true) -> if (data.autoSave)?
I think you are also saying:
if (data.autoSave==true) -> return data.autoSave;?

This latter assumes that autosaving should be done if Autosave.M3S exists (even if it is not an autosave) is that okay? If so I may not need to bump the version number or even include an autoSave flag since just checking for the autosave file is sufficient.

Reply:
The current code after reading the save is strictly equivalent to `return data.autoSave;`